### ISSUE TITLE
Compare VODE to VODE90. Determine if a switch to VODE90 is in order.

### ISSUE THREAD

**Issue by dwillcox** [starkiller-astro/Microphysics#52]
At the moment, we have two VODE-style integrators, VODE and VODE90.

Earlier testing indicates they yield identical integration answers but there may be performance differences. We should compare the performance of each and determine whether it is worth switching to VODE90.

Issues to consider:

- VODE90's use of derived types may slow down the GPU. We may need to refactor a bit to eliminate derived types.
- VODE90 seems a bit slower than VODE, but we should check this and figure out why.

**Comment by dwillcox** [comment]
The following test is on `merger` on development branch of Microphysics using `unit_test/test_react` with the following compilation flags:

`make -j NDEBUG=t MPI= OMP= COMP=gfortran NETWORK_DIR=aprox13 EOS_DIR=helmholtz`

with `INTEGRATOR_DIR=VODE90` or `INTEGRATOR_DIR=VODE`.

Due to the recent updates to VODE enforcing limits on mass fractions and T, I'm using commit `d98ac7843db80894f78a955f62ca6c3ea282209b` from September 12, 2017. The AMReX git commit is `266f31c00dfacb03b58a551006dce04ee6cc60a7` from September 18, 2017.

**VODE**

```
[dwillcox@merger test_react]((d98ac78...))$ ./main.Linux.gfortran.exe inputs_aprox13
 MPI initialized with            1  MPI processes
 MPI initialized with            1  threads
 
 Initializing Helmholtz EOS and using Coulomb corrections.
 
 small_temp =    100000.00000000000     
 small_dens =    1.0000000000000000     
 Execution time:    156.29828095436096     
 RHS stats:
   min:            6
   avg:         1522
   max:        23965
```

**VODE90**

```
[dwillcox@merger test_react]((d98ac78...))$ ./main.Linux.gfortran.exe inputs_aprox13
 MPI initialized with            1  MPI processes
 MPI initialized with            1  threads
 
 Initializing Helmholtz EOS and using Coulomb corrections.
 
 small_temp =    100000.00000000000     
 small_dens =    1.0000000000000000     
 Execution time:    160.93007707595825     
 RHS stats:
   min:            6
   avg:         1522
   max:        23965
```

**fcompare**

`fcompare` reports no differences in the plotfiles written by `test_react` with "PLOTFILES AGREE"

All this suggests that on the CPU the two are pretty comparable with a timing difference relative to VODE of 2.9%.

**Comment by dwillcox** [comment]
I've repeated this test 9 more times for a total of 10, and the average times plus or minus a standard deviation for each integrator are as follows:

VODE: 156.2 +/- 1.2 seconds

VODE90: 159.7 +/- 1.3 seconds

Relative difference wrt VODE: 2.25%

Timing spreadsheet: [VODE-vs-VODE90-aprox13.pdf](https://github.com/starkiller-astro/Microphysics/files/1316698/VODE-vs-VODE90-aprox13.pdf)

**Comment by maxpkatz** [comment]
Can we get this in a branch and then run the test suite on it to check differences?

**Comment by dwillcox** [comment]
Not quite yet, a few days ago I realized the timing above is for the original VODE90 revisions (on `development`, before I made the changes needed to make it operable with CUDA Fortran.

When I've run timing between the versions of VODE90 and VODE on the `cudadevice` branch in serial mode on the CPU I'm getting larger timing differences of ~7% and need to determine why. I should be able to look into this more today.
